analysis/runtime/src/runtime/skip.rs

```diff
@@ -45,15 +45,11 @@ pub(super) fn skip_event(event: Event, reason: SkipReason) {
     use SkipReason::*;
     match reason {
         BeforeMain => {
+            // # Signal-safety: atomic increments are safe.
             EVENTS_SKIPPED_BEFORE_MAIN.fetch_add(1, Ordering::Relaxed);
         }
         AfterMain => {
-            // This is after `main`, so it's safe to use things like `eprintln!`,
-            // which uses `ReentrantMutex` internally, which may use `pthread` mutexes.
-            WARNED_AFTER_MAIN.get_or_init(|| {
-                eprintln!("skipping {reason}");
-            });
-            eprintln!("skipped event after `main`: {:?}", event.kind);
```

Comment:
Shouldn't we still be able to print to the unbuffered `stderr` if we make no allocations?  It's useful to have at least some sort of warning here.

Reply:
I initially thought `write` wasn't signal-safe, but https://man7.org/linux/man-pages/man7/signal-safety.7.html says it is. The first `eprintln!` is probably safe to call (I'd need to check its implementation), but the second one probably isn't because the formatter allocates.

Reply:
The second one is more useful and they both format an argument.  The second one can be changed to also `eprintln!("{}", event.kind.as_str())` or something like that, though.

Reply:
If `write` wasn't async-signal-safe that would be very bad.  Then you can't do much at all in the signal handler (like the pipe trick).

Reply:
I put the old code back in with some changes. The `eprintln!` is still problematic, but I don't see a way to get rid of it.